dialog/dialog/spiders/dialog.py

```diff
@@ -0,0 +1,93 @@
+import scrapy
+import re
+from dialog.items import Publication
+
+INDEX_URL = 'http://dialogsocial.gov.ro/categorie/proiecte-de-acte-normative/'
+
+DOC_EXTENSIONS = [
+    ".docs", ".doc", ".txt", ".crt", ".xls",
+    ".xml", ".pdf", ".docx", ".xlsx",
+]
+
+def text_from(sel):
+    return sel.xpath('string(.)').extract_first().strip()
+
+def guess_publication_type(text):
+    text = text.lower()
+    text = re.sub(r'[șş]', 's', text)
+    text = re.sub(r'[țţ]', 't', text)
+    text = re.sub(r'[ăâ]', 'a', text)
+    text = re.sub(r'[î]', 'i', text)
+    rules = [
+        ("lege", "LEGE"),
+        ("hotarare de guvern", "HG"),
+        ("hotarare a guvernului", "HG"),
+        ("ordonanta de guvern", "OG"),
+        ("ordonanta de urgenta", "OUG"),
+        ("ordin de ministru", "OM"),
+        ("ordinul", "OM"),
+    ]
+    for substr, publication_type in rules:
+        if substr in text:
```

Comment:
This should be some sort of eager match, taking into account all possibilities (no early bailout).
For example, "Hotarare de guvern privind legea X" would return the wrong type.